packages/integrations/components/coral/withPico.js

```diff
@@ -0,0 +1,107 @@
+import React, { useState, useCallback, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import CoralEmbed from './index';
+import {
+  actionVerifyPicoUser,
+  actionReceiveCoralLogoutRequest,
+} from '../../actions';
+import { tokenSelector } from '../../selectors/coralSelector';
+
+const withPico = (ChildComponent) => (props) => {
+  const [verificationRequestSent, setRequestStatus] = useState(false);
+
+  const dispatch = useDispatch();
+  // Define a function to dispatch Pico user verification requests.
+  const dispatchVerificationRequest = useCallback(
+    (user) => {
+      // Update the request status variable so that multiple verification
+      // requests are prevented.
+      setRequestStatus(true);
+      // Dispatch the action.
+      return dispatch(actionVerifyPicoUser(user));
+    },
+    [dispatch]
+  );
+  // Define a function to dispatch logout requests via SSO.
+  const dispatchLogoutRequest = useCallback(
+    () => dispatch(actionReceiveCoralLogoutRequest()),
+    [dispatch]
+  );
+
+  useEffect(() => {
+    const observerHandler = () => {
+      const signalNode = document.getElementById('PicoSignal-container');
+      // Only mount the observer if the PicoSignal button exists in the DOM and
+      // the Pico data attributes have been appended to the element.
+      if (
+        signalNode &&
+        null !== signalNode.getAttribute('data-pico-email') &&
+        false === verificationRequestSent
+      ) {
+        // Define the observer.
+        const observer = new MutationObserver((mutations) => {
```

Comment:
Not necessary for this PR maybe, but I'd love to see an abstraction of a `MutationObserver` hook (maybe `useMutation`?) in irving core. We have one for intersection observers currently.

Reply:
I'm definitely for that. I'll put together an integration ticket for that work.